fairseq/data/language_pair_dataset.py

```diff
@@ -139,6 +139,10 @@ def valid_size(self, index, max_positions):
     def _get_max_positions(self, max_positions):
         if max_positions is None:
             return self.max_source_positions, self.max_target_positions
+        # If max_position is not a sequence (possibly from a monolingual model)
+        # then use the same upper bound for source & target
+        if not hasattr(max_positions, '__len__'):
```

Comment:
can we check if its not type(tuple) instead ?

Reply:
sorry i meant check if type is int here

Reply:
max_positions was actually a float, I think this check is more general as it capture both tuple and list...